xonsh/environ.py

```diff
@@ -81,18 +162,127 @@ def format_prompt(template=default_prompt):
               BOLD_INTENSE, BACKGROUND_INTENSE
     + NO_COLOR -- Resets any previously used color codes
     """
+    def __init__(self, prompt_template, *args, **kwargs):
+        super(DefaultPrompt, self).__init__(prompt_template, *args, **kwargs)
+        self._env = builtins.__xonsh_env__
+
+        self.update(TERM_COLORS)
+        self.update(dict(
```

Comment:
I don't think the call to dict is needed here.  Update accepts the same arguments as dict.